server/cluster.go

```diff
@@ -1169,8 +1169,8 @@ func (c *RaftCluster) AllocPeer(storeID uint64) (*metapb.Peer, error) {
 
 // OnStoreVersionChange changes the version of the cluster when needed.
 func (c *RaftCluster) OnStoreVersionChange() {
-	c.Lock()
-	defer c.Unlock()
+	c.RLock()
+	defer c.RUnlock()
```

Comment:
Is it really safe? I see two functions have write semantic: `c.opt.SetClusterVersion()` and `c.opt.Persist()`. And there is no code or comments indicating they are thread-safe.